core/txpool/legacypool/legacypool.go

```diff
@@ -672,25 +672,30 @@ func (pool *LegacyPool) validateTx(tx *types.Transaction, local bool) error {
 			}
 			return have, math.MaxInt
 		},
-		ExistingExpenditure: func(addr common.Address) *big.Int {
+		ExistingExpenditure: func(addr common.Address) (*big.Int, *big.Int) {
 			if list := pool.pending[addr]; list != nil {
-				return list.TotalCostFor(tx.FeeCurrency()).ToBig()
+				return list.TotalCostFor(tx.FeeCurrency()).ToBig(), list.TotalCostFor(tx.FeeCurrency()).ToBig()
```

Comment:
Shouldn't the native expenditure query for the `nil` fee currency expenditure in the list?

```suggestion
				return list.TotalCostFor(tx.FeeCurrency()).ToBig(), list.TotalCostFor(nil).ToBig()
```

If a new transaction is added to the pool, the write operation to the `totalCostVar` makes a distinction between the two:

https://github.com/celo-org/op-geth/blob/3e4fa03d1a35667acc7c5e7ff24ddfd5b9fc09b4/core/txpool/legacypool/list.go#L360-L361

Reply:
Actually not `nil` but `getCurrencyKey(nil) == ZeroAddress`. But your main point is right. Which shows that the testing of this is in a bad state at the moment.

Reply:
My bad, both are identical, since `TotalCostFor` does `getCurrencyKey(feeCurrency)` internally.

Reply:
Done in https://github.com/celo-org/op-geth/pull/118/commits/0f8aebd7a0f16285b3f2d56d135e09c92182cee8